homeassistant/components/usb/flow.py

```diff
@@ -0,0 +1,50 @@
+"""The USB Discovery integration."""
+from __future__ import annotations
+
+from collections.abc import Coroutine
+from typing import Any, TypedDict
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+
+
+class USBFlow(TypedDict):
+    """A queued usb discovery flow."""
+
+    domain: str
+    context: dict[str, Any]
+    data: dict
+
+
+class FlowDispatcher:
+    """Dispatch discovery flows."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Init the discovery dispatcher."""
+        self.hass = hass
+        self.pending_flows: list[USBFlow] = []
+        self.started = False
+
+    @callback
+    def async_start(self, *_: Any) -> None:
+        """Start processing pending flows."""
+        self.started = True
+        self.hass.loop.call_soon(self._async_process_pending_flows)
```

Comment:
What's the reason we use the loop api directly? We probably need a comment for that if it's needed.

Reply:
I'm been working on python-zeroconf too much and been using them directly. I'll switch it to async_add_job